pkg/envoy/cds/response.go

```diff
@@ -16,74 +15,46 @@ import (
-	}
+	outboundMeshTrafficClusterConfigs := meshCatalog.GetOutboundMeshTrafficPolicy(proxy.Identity).ClustersConfigs
+	cb.SetOutboundMeshTrafficClusterConfigs(outboundMeshTrafficClusterConfigs)
 
-	// Build local clusters based on allowed inbound traffic policies
 	proxyServices, err := proxyRegistry.ListProxyServices(proxy)
 	if err != nil {
 		log.Error().Err(err).Str(errcode.Kind, errcode.GetErrCodeWithMetric(errcode.ErrFetchingServiceList)).
 			Str("proxy", proxy.String()).Msg("Error looking up MeshServices associated with proxy")
 		return nil, err
 	}
-	inboundMeshTrafficPolicy := meshCatalog.GetInboundMeshTrafficPolicy(proxy.Identity, proxyServices)
-	if inboundMeshTrafficPolicy != nil {
-		clusters = append(clusters, localClustersFromClusterConfigs(inboundMeshTrafficPolicy.ClustersConfigs)...)
-	}
 
-	// Add egress clusters based on applied policies
-	if egressTrafficPolicy, err := meshCatalog.GetEgressTrafficPolicy(proxy.Identity); err != nil {
-		log.Error().Err(err).Msgf("Error retrieving egress policies for proxy with identity %s, skipping egress clusters", proxy.Identity)
-	} else {
-		if egressTrafficPolicy != nil {
-			clusters = append(clusters, getEgressClusters(egressTrafficPolicy.ClustersConfigs)...)
-		}
-	}
+	inboundMeshTrafficClusterConfigs := meshCatalog.GetInboundMeshTrafficPolicy(proxy.Identity, proxyServices).ClustersConfigs
+	cb.SetInboundMeshTrafficPolicy(inboundMeshTrafficClusterConfigs)
 
-	outboundPassthroughCluser, err := getOriginalDestinationEgressCluster(envoy.OutboundPassthroughCluster, nil)
+	egressTrafficPolicy, err := meshCatalog.GetEgressTrafficPolicy(proxy.Identity)
 	if err != nil {
-		log.Error().Err(err).Str(errcode.Kind, errcode.ErrGettingOrgDstEgressCluster.String()).
-			Msgf("Failed to passthrough cluster for egress for proxy %s", envoy.OutboundPassthroughCluster)
-		return nil, err
+		log.Error().Err(err).Msgf("Error retrieving egress policies for proxy with identity %s, skipping egress clusters", proxy.Identity)
 	}
+	cb.SetEgressTrafficClusterConfigs(egressTrafficPolicy.ClustersConfigs)
 
-	// Add an outbound passthrough cluster for egress if global mesh-wide Egress is enabled
-	if cfg.GetMeshConfig().Spec.Traffic.EnableEgress {
-		clusters = append(clusters, outboundPassthroughCluser)
-	}
+	cb.SetEgressEnabled(cfg.GetMeshConfig().Spec.Traffic.EnableEgress)
 
 	// Add an inbound prometheus cluster (from Prometheus to localhost)
 	if enabled, err := meshCatalog.IsMetricsEnabled(proxy); err != nil {
 		log.Warn().Str("proxy", proxy.String()).Msg("Could not find pod for connecting proxy, no metadata was recorded")
 	} else if enabled {
-		clusters = append(clusters, getPrometheusCluster())
+		cb.SetMetricsEnabled(enabled)
 	}
 
 	// Add an outbound tracing cluster (from localhost to tracing sink)
-	if cfg.GetMeshConfig().Spec.Observability.Tracing.Enable {
-		clusters = append(clusters, getTracingCluster(cfg))
+	if enabled := cfg.GetMeshConfig().Spec.Observability.Tracing.Enable; enabled {
```

Comment:
nit: you don't need the `enabled` var, since you're in a truthy block, you're guaranteed enabled is true, and it's a *small* amount of less cognitive overhead to just use the true value

although I'm proposing getting rid of even storing the enabled var altogether in the following comment